pkg/operator/controller/ingress/status.go

```diff
@@ -674,6 +674,29 @@ func computeIngressProgressingCondition(conditions []operatorv1.OperatorConditio
 				condition.Reason = "ScopeChanged"
 				condition.Message = message
 				condition.Status = operatorv1.ConditionTrue
+			} else {
+				switch platform.Type {
+				case configv1.AWSPlatformType:
+					if _, ok := service.Annotations[AWSLBTypeAnnotation]; ok {
+						if (service.Annotations[AWSLBTypeAnnotation] == AWSNLBAnnotation) &&
+							(ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != operatorv1.AWSNetworkLoadBalancer) ||
```

Comment:
Thanks for trying out this change. Do you really need the first stanza that checks for `nil` on `ProvidedParameters` and `AWS` if you rearrange properly?
`(ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != operatorv1.AWSNetworkLoadBalancer)`?

Couldn't you just do (in this order specifically):
```
if (service.Annotations[AWSLBTypeAnnotation] == AWSNLBAnnotation) &&
(ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters == nil ||
ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS == nil ||
ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != operatorv1.AWSNetworkLoadBalancer) {
```

The same short circuiting you are using with "don't check `ProviderParameters.AWS.Type` if a prior `&&` is false, is the same with `||` if it is true. It won't attempt to evaluate the next `||` or conditions. Based on the fact you are using `if` conditional short-circuiting, I don't see a reason not to use it with `||`. It is just as `nil` safe as this. Thoughts?

Reply:
We need to check if `ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != operatorv1.AWSNetworkLoadBalancer)`  is not set to NLB(generally for classic LB this is not set but there might be a case where Classic is set when someone wants to specify params in the provider) so for doing that you will need to check if ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil to ensure go does not throw nil error.

I don't think `OR` and  `== nil` will help.

Also we will also need to check the condition if  `(ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters == nil || ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS == nil)`

as for having classic LB  in ingresscontroller it is not mandatory to have the ProviderParameters set in the ingress controller.

Reply:
Sorry, I don't mean to drag this one out, but I want to make sure we are on the same page. Totally agree with what you said, but just pointing out it could be simplified which makes it a bit more readable.

So do you agree that:
```
if (a.b != nil && a.b.c != nil && a.b.c.d == "x") ||
    (a.b == nil || a.b.c == nil) {
    do something
}
```
the same as this:
```
if a.b == nil || a.b.c == nil || a.b.c.d == "x" {
    do something
}
```
?

Reply:
@gcs278  Agree. Thanks. PTAL at the changes I made as per your request.